hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/HazelcastCacheRegionFactory.java

```diff
@@ -48,8 +45,9 @@ public HazelcastCacheRegionFactory(final Properties properties) {
 
     public CollectionRegion buildCollectionRegion(final String regionName, final Properties properties,
                                                   final CacheDataDescription metadata) throws CacheException {
+        /* Collection regions are never versioned, so pass in null for metadata */
         return new HazelcastCollectionRegion<IMapRegionCache>(instance, regionName, properties, metadata,
-                new IMapRegionCache(regionName, instance, properties, metadata));
+                new IMapRegionCache(regionName, instance, properties, null));
```

Comment:
Why do you override `metadata` with null? Can somebody not provide metadata for `CollectionRegion`?